kong/globalpatches.lua

```diff
@@ -14,6 +14,27 @@ return function(options)
   options = options or {}
   local meta = require "kong.meta"
 
+  do -- make cjson encode numbers with a precision of up to 16 decimals
+    local cjson = require "cjson"
+    local cjson_safe = require "cjson.safe"
+    local native_cjson_new = cjson.new
+    local native_cjson_safe_new = cjson_safe.new
+
+    cjson.encode_number_precision(16)
+    cjson_safe.encode_number_precision(16)
+
+    cjson.new = function()
+      local json = native_cjson_new()
+      json.encode_number_precision(16)
+      return json
+    end
+
+    cjson_safe.new = function()
+      local json = native_cjson_safe_new()
+      json.encode_number_precision(16)
+      return json
+    end
+  end
```

Comment:
Is it possible for us to not make this global? Can we only address the issue related to Admin API display and input?

Reply:
Technically it's possible, it just feels like our behavior for this issue would be confusing if we just fixed the admin API. For example, if the user configures the serverless plugin themselves and uses `cjson.new`, the precision is still 14 bits. Similarly, Valut has this problem, which I personally don't feel is very good.

Reply:
Let's not get ahead of ourself, most of the places won't run into this issue anyway and doing a global patch sounds too excessive. Could you provide an alternative patch just for addressing [FTI-5515](https://konghq.atlassian.net/browse/FTI-5515) and compare?

Reply:
Also, if we really must change the default, changing it using a patch against https://github.com/openresty/lua-cjson/blob/master/lua_cjson.c#L82 would be a much better idea than introducing another global patch into Kong, these kind of patches should be ideally as little as possible before getting impossible to maintain. But please analyze my comment above first.

Reply:
Sounds good,  will do.